### ISSUE TITLE
Consider deduplicating override.typaram.invalid against override.param.invalid/override.return.invalid for the same occurrence

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#1973]
`BaseTypeVisitor.OverrideChecker.checkTypeParameterBounds` (added in #1961) and the pre-existing per-occurrence checks (`checkParameters`/`checkReturn`, via `isParameterOverrideValid`/`isReturnOverrideValid`'s own type-variable fallback) can both independently reject the same underlying mistake, producing two diagnostics for one root cause. This happens whenever a type parameter's *own declared bound* differs between the overriding and overridden methods, and that type parameter is also used as a *bare* (unannotated) occurrence in the parameter or return type:

```java
static class Super {
    <T extends @Nullable Object> void consumeList(List<T> p) {}
}

static class Sub extends Super {
    @Override
    // :: error: (override.typaram.invalid)
    // :: error: (override.param.invalid)
    <T extends @NonNull Object> void consumeList(List<T> p) {}
}
```

(`SubNestedTypeParam` in `checker/tests/nullness/OverrideTypeParamBound.java` on the #1961 branch; several other cases in the same file's regression matrix have the identical pattern.)

This is intentional for now -- both diagnostics are correct, just redundant -- but it's worth deciding whether `checkTypeParameterBounds` should skip a type parameter that `checkParameters`/`checkReturn` already rejected an occurrence of. Doing so would require `checkTypeParameterBounds` to know what the earlier checks found, which is a real ordering dependency between the two (`checkTypeParameterBounds` already runs after `checkReturn`/`checkParameters` in `checkOverride`, so the information is available in principle, but nothing currently threads it through). It would also fix #1972 (a KeyFor-specific instance of the same pattern) as a side effect.

Not filing this as "the fix is X" -- it's a design question first: is the double diagnostic actually a problem in practice (confusing to users, or just harmless redundancy), and if so, is the ordering-dependency machinery worth adding for it? #1961's own review deliberately didn't build this speculatively, per this project's general policy against adding coupling without a concrete need.
